synchronizer/synchronizer.go

```diff
@@ -110,41 +121,26 @@ func (s *ClientSynchronizer) Sync() error {
 					continue
 				}
 			}
-			if waitDuration != s.cfg.SyncInterval.Duration {
-				latestsequencedBatchNumber, err := s.etherMan.GetLatestBatchNumber()
-				if err != nil {
-					log.Warn("error getting latest sequenced batch in the rollup. Error: ", err)
-					continue
-				}
-				// Check latest Synced Batch
-				dbTx, err := s.state.BeginStateTransaction(s.ctx)
-				if err != nil {
-					log.Fatalf("error creating db transaction to get latestSyncedBatch. error: %s", err.Error())
-				}
-				latestSyncedBatch, err := s.state.GetLastBatchNumber(s.ctx, dbTx)
-				errC := dbTx.Commit(s.ctx)
-				if errC != nil {
-					log.Errorf("error committing dbTx, err: %s", errC.Error())
-					rollbackErr := dbTx.Rollback(s.ctx)
-					if rollbackErr != nil {
-						log.Fatalf("error rolling back state. RollbackErr: %s, err: %s",
-							rollbackErr.Error(), errC.Error())
-					}
-					log.Fatalf("error committing dbTx, err: %s", errC.Error())
-				}
+			latestSequencedBatchNumber, err := s.etherMan.GetLatestBatchNumber()
+			if err != nil {
+				log.Warn("error getting latest sequenced batch in the rollup. Error: ", err)
+				continue
+			}
+			latestSyncedBatch, err := s.state.GetLastBatchNumber(s.ctx, nil)
+			if err != nil {
+				log.Warn("error getting latest batch synced. Error: ", err)
+				continue
+			}
+			if latestSyncedBatch >= latestSequencedBatchNumber {
+				log.Info("L1 state fully synchronized")
+				err = s.syncTrustedState(latestSyncedBatch)
```

Comment:
Maybe including this into the else, you could replicate the same logic.
```
if waitDuration != s.cfg.SyncInterval.Duration {

} else {
      err = s.syncTrustedState(latestSyncedBatch)
}
```

Reply:
We should wait until the l1 synchronization is done.The waitDuration is 0 at the beginning. So if it is equal to the value in the config file, l1 is synced and we could start syncing the trusted state. We could have problems with s.cfg.SyncInterval.Duration = 0s
We will syncing l1 every waitDuration but we want check if it is updated because it is.

Reply:
`waitDuration != s.cfg.SyncInterval.Duration` means the variable used to control the interval between sync cycles and the parameter configured to define this interval are different, this doesn't mean `L1 was fully synchronized`

What defines if the L1 was fully synced is checking the L2 last batch against the last batch in the rollup.